### PR TITLE
use language order from the client's Accept-Language: HTTP header

### PR DESCRIPTION
**Pull request by errotan**

### CODE REVIEW

**Comment by piotrgolasz** [comment]
Some unit tests to that?

**Comment by errotan** [comment]
Added the unit tests.

File: system/classes/Kohana/HTTP/Header.php [hunk 1]
```diff
@@ -811,6 +816,81 @@ public function accepts_language_at_quality($language, $explicit = FALSE)
...
+			$this->_accept_language_list = HTTP_Header::_parse_language_header_as_list($language_header);
+		}
+
+		$new_order = [];
+
+		foreach ($this->_accept_language_list as $accept_language)
+		{
+			foreach ($languages as $key => $language)
+			{
+				if (($explicit AND $accept_language == $language) OR
```
Comment (knbv) [inline system/classes/Kohana/HTTP/Header.php]: Is it possible to use a non-strict comparison here?

Comment (neo22s) [reply system/classes/Kohana/HTTP/Header.php]: then its ok or not?

Comment (knbv) [reply system/classes/Kohana/HTTP/Header.php]: I think that you need to use a strict comparison in the following lines: 875, 876

The fact is that a non-strict comparison can be unsafe. Or is it not important in this case?

**Pull request closed without merging**